### DISCUSSION TITLE
How to pick a function type for my data

### DISCUSSION THREAD

**Discussion by amna-ukaea** [idaholab/moose#25254] [Q&A General]
Hi all,
I have data that is heat at different x points at different times to inform a boundary condition. So if I were to put it in columns it would looks like:
time x heat
0 x0 heat 1
0 x1 heat 2
0 x2 heat 3
1 x0 heat 4
1 x1 heat 5
1 x2 heat 6

At the moment I'm using a vector processor of type CSVreader to read the data and a function of type NearestReporterFunction and a BC of type FunctionDirichletBC to use the funnction. It isn't working and the error is "No valid AXIS lines found by GriddedData"

Any ideas on how I can fix this either changing the data structure or the input file?

Thanks in advnace

**Comment by GiudGiud** [comment]
Hello

For data depending on time and a spatial position, we can also use either of these two
https://mooseframework.inl.gov/source/functions/PiecewiseMulticonstant.html
https://mooseframework.inl.gov/source/functions/PiecewiseMultilinear.html
They will require some modification to the data structure to add some of these AXIS X Y Z T columns

I m puzzled by your error
`No valid AXIS lines found by GriddedData` is returned by GriddedData which is used by the functions I linked to but not by any of the objects you mention.
Are you sure it doesnt come from some extraneous object in your input file?

also what is NearestReporterFunction? I see a NearestReporterCoordinatesFunction in the optimization module

Guillaume
